src/Rule/Nested.php

```diff
@@ -169,4 +168,36 @@ private function fetchOptions(array $rules): array
 
         return $result;
     }
+
+    private static function calculateErrorKey(Rules $aggregatedRule, string $valuePath, string $key): string
+    {
+        if (!self::canConcatenateErrorKey($aggregatedRule, $key)) {
+            return $valuePath;
+        }
+
+        return $valuePath . Result::ERROR_KEY_SEPARATOR . $key;
+    }
+
+    private static function canConcatenateErrorKey(Rules $aggregatedRule, string $key): bool
+    {
+        if (!self::isPositiveInteger($key)) {
```

Comment:
Seems, we can check is_int($key) in `foreach ($itemResult->getErrors() as $key => $error) {` above.

Reply:
You mean use different logic for detecting a positive integer? Or move it to the loop? Or both?  Sorry, but in case of the latter, how it will improve the current code? Moving just this check is not enough too. What's the reason of extracting it?

Reply:
Refactored. Please check.

Reply:
:heavy_check_mark:

Reply:
No longer actual too.